docs/core/transactions.md

```diff
@@ -0,0 +1,84 @@
+
+A transaction is created through one of the possible [interfaces](#interfaces). In the process, two contexts and an array of [messages](#messages) are created, which are then used to [generate](#transaction-generation) the transaction itself. The actual state changes triggered by transactions are enabled by the [handlers](#handlers). The rest of the document will describe each of these components, in this order.
+
+## CLI and REST Interfaces
+
+The SDK uses several tools for building [interfaces](./interfaces.md) through which users can create transactions through the command-line or HTTP Requests. Application developers create entrypoints to the application by creating a [command-line interface](./interfaces.md#cli) or [REST interface](./interfaces.md#rest), typically found in the application's `/cmd` folder. These interfaces allow users to interact with the application through command-line or through HTTP requests.
+
+In order for module messages to be utilized in transactions created through these interfaces, module developers must also specify possible user [interactions](../modules/interfaces.md), typically in the module's `/client` folder. For the [command-line interface](../modules/interfaces.md#cli), module developers create subcommands to add as children to the application top-level transaction command `TxCmd`. For [HTTP requests](../modules/interfaces.md#rest), module developers specify acceptable request types, register REST routes, and create HTTP Request Handlers.
+
+When users interact with the application's interfaces, they invoke the underlying modules' handlers or command functions, directly creating messages.
+
+## Messages
+
+**Messages** are module-specific objects that trigger state transitions within the scope of the module they belong to. Module developers define the messages for their module by implementing the [`Msg`](https://github.com/cosmos/cosmos-sdk/blob/master/types/tx_msg.go#L10-L31) interface, and also define a [`Handler`](../building-modules/handler.md) to process them. Messages in a module are typically defined in a `msgs.go` file (though not always), and one handler with multiple functions to handle each of the module's messages is defined in a `handler.go` file.
+
+Note: module messages are not to be confused with [ABCI Messages](https://tendermint.com/docs/spec/abci/abci.html#messages) which define interactions between the Tendermint and application layers. While ABCI messages such as CheckTx and DeliverTx contain Transactions, which contain module Messages, they are not to be confused with the module level messages themselves
+
+The [`Msg`](https://github.com/cosmos/cosmos-sdk/blob/master/types/tx_msg.go#L10-L31) interface has five required functions.
```

Comment:
No need to expand on the interface here, just link to `../building-modules/messages-and-queries.md#messages`